terraform/rob-cos/README.md

```diff
@@ -0,0 +1,78 @@
+# Terraform module for `rob-cos`
+
+This is a Terraform module facilitating the deployment of the COS for Devices project,
+using the [Terraform Juju provider](https://github.com/juju/terraform-provider-juju/).
+For more information,
+refer to the provider [documentation](https://registry.terraform.io/providers/juju/juju/latest/docs).
+
+> [!CAUTION]
+> This module is not intended to be deployed in production.
+> It rather is a demonstrator as well as a starting point for developing a production-grade configuration.
+
+## Usage
+
+Users should ensure that Terraform is aware of the `juju_model` dependency of the charm module.
+
+To deploy this module with its needed dependency, you can run:
+
+```bash
+terraform apply -var="model=<K8S_MODEL_NAME>"
```

Comment:
Shouldn't this model be capable of including the rob-cos-microceph terraform?

Reply:
I could bloat it with the microceph bits and make them all optional but them this doesn't scale (with other optional features we might want to add). Instead, this model deploys all of rob-cos but the storage back-end.
You'll find a model that deploys both rob-cos and the microceph backend in the `rob-cos-microceph` folder.